DOCUMENTATION.md

```diff
@@ -12,48 +12,54 @@ The main goal of this platform is to act as a web-based alternative for Github d
 - [Add a new branch](#add-new-branch)
  
 ## Requirements
-1. [Node JS](https://nodejs.org/en/) (Version 12.0+)
-2. [Git](https://git-scm.com/) (Version 2.20+)
+1. [Git](https://git-scm.com/) (Version 2.20+)
+
+If you wish to build the application from source then install,
+   - [Node JS](https://nodejs.org/en/) (Version 12.0+)
+   - [go](https://golang.org/dl/)
+
 ## Download Options
 - **Option-1:** Directly clone the repo from GitHub
-`git clone https://github.com/neel1996/gitconvex-package`
+`git clone https://github.com/neel1996/gitconvex`
 
-- **Option-2:** Downloading the zip file from the [releases](https://github.com/neel1996/gitconvex-package/releases).
+- **Option-2:** If you are into docker, then there is also a docker image available for gitconvex
+`docker pull itassistors/gitconvex`
 
-- **Option-3:** GitConvex is also available on `npm`. Install the package globally to run it directly from the command line
-`npm i -g @itassistors/gitconvex`
+- **Option-3:** Downloading the zip file from the [releases](https://github.com/neel1996/gitconvex/releases).
 
-This will install **GitConvex** as a global module and it can be started straight away from the command line with `gitconvex` command
 ```
-$ gitconvex
-  
-INFO: Checking for config file
-INFO: Config file is present
-INFO: Reading from config file /usr/lib/node_modules/@itassistors/gitconvex/env_config.json
-GitConvex API connected!
-  
-Checking data file availability...
-INFO: Data file /usr/lib/node_modules/@itassistors/gitconvex/database/repo-datastore.json is present and it will be used as the active data file!
-You can change this under the settings menu
-GitConvex is running on port 9001
-Open http://localhost:9001/ to access GitConvex
+## Extract the downloaded zip file and execute the commands
+
+$ cd gitconvex
+
+# for Mac & Linux
+$ make build
+
+# for Windows
+$ ./make.bat build
+
 ```
+
 ## Setup
+$ git clone https://github.com/neel1996/gitconvex.git
+<br>
+$ cd gitconvex
+
+# For Mac & Linux
+$ make build
 
-If either download **Option-1** or **Option-2** is opted, then the following steps need to be followed to setup GitConvex
-1. For installing all dependencies,
-`` $npm install ``
+# For Windows
+$ ./make.bat build
 
-2. To start the server, either use normal node command
-`` $node server.js ``
-or use `pm2` by downloading it from npm - `npm i -g pm2` and start the module by executing the following command,
-`pm2 start ecosystem.config.js`
+## After build completion...
+$ ./dist/gitconvex-server
```

Comment:
replace gitconvex-server with `gitconvex` for IX and `gitconvex.exe` for windows